src/com/google/enterprise/adaptor/fs/WindowsFileDelegate.java

```diff
@@ -551,8 +596,14 @@ public void callback(int errorCode, int nBytesTransferred,
                 + "callback for {0}.", watchPath);
           }
           continue;
+        } else if (waitResult == Kernel32.WAIT_TIMEOUT) {
+          if (logging) {
+            log.log(Level.FINER, "Timed out waiting for notifications from {0}."
+                + " Retrying.", watchPath);
+          }
+          continue;
         } else if (waitResult == WinBase.WAIT_OBJECT_0) {
-          log.log(Level.FINE, "Terminate event has been set, ending file "
+          log.log(Level.FINE, "Terminate event has been received, ending file "
               + "monitor for {0}.", watchPath);
           return;
         } else {
```

Comment:
You kept the else if (WAIT_FAILED) then bail out above, with no else (which turns into an implied retry with no logging), and the single else lumping them together for a retry here. Is that asymmetry worth a comment while we're thinking about it?